pgmpy/inference/ExactInference.py

```diff
@@ -40,10 +40,7 @@ def _variable_elimination(self, variables, operation, evidence=None, elimination
 
         # Dealing with the case when variables is not provided.
         if not variables:
-            all_factors = []
-            for factor_li in self.factors.values():
-                all_factors.extend(factor_li)
-            return set(all_factors)
+            return {}
```

Comment:
@lohani2280 I can't understand this change. Elaborate please ? And I think the tests are failing because of this.

Reply:
Sorry, my bad. Hadn't read the whole issue.